pandas/tests/io/test_excel.py

```diff
@@ -2260,54 +2122,28 @@ def test_column_format(self):
             assert read_num_format == num_format
 
 
-class TestOpenpyxlTests_NoMerge(ExcelWriterBase):
-    ext = '.xlsx'
-    engine_name = 'openpyxl'
-    check_skip = staticmethod(_skip_if_no_openpyxl)
-
-    # Test < 0.13 non-merge behaviour for MultiIndex and Hierarchical Rows.
-    merge_cells = False
-
-
-class TestXlwtTests_NoMerge(ExcelWriterBase):
-    ext = '.xls'
-    engine_name = 'xlwt'
-    check_skip = staticmethod(_skip_if_no_xlwt)
-
-    # Test < 0.13 non-merge behaviour for MultiIndex and Hierarchical Rows.
-    merge_cells = False
-
-
-class TestXlsxWriterTests_NoMerge(ExcelWriterBase):
-    ext = '.xlsx'
-    engine_name = 'xlsxwriter'
-    check_skip = staticmethod(_skip_if_no_xlsxwriter)
-
-    # Test < 0.13 non-merge behaviour for MultiIndex and Hierarchical Rows.
-    merge_cells = False
-
-
 class TestExcelWriterEngineTests(object):
 
-    def test_ExcelWriter_dispatch(self):
-        with tm.assert_raises_regex(ValueError, 'No engine'):
-            ExcelWriter('nothing')
-
-        try:
-            import xlsxwriter  # noqa
-            writer_klass = _XlsxWriter
-        except ImportError:
-            _skip_if_no_openpyxl()
-            writer_klass = _OpenpyxlWriter
-
-        with ensure_clean('.xlsx') as path:
+    @pytest.mark.parametrize('klass,ext', [
+        pytest.param(_XlsxWriter, '.xlsx', marks=pytest.mark.skipif(
+            not td.safe_import('xlsxwriter'), reason='No xlsxwriter')),
+        pytest.param(_OpenpyxlWriter, '.xlsx', marks=pytest.mark.skipif(
+            not td.safe_import('openpyxl'), reason='No openpyxl')),
+        pytest.param(_XlwtWriter, '.xls', marks=pytest.mark.skipif(
+            not td.safe_import('xlwt'), reason='No xlwt'))
+    ])
+    def test_ExcelWriter_dispatch(self, klass, ext):
```

Comment:
This test was pretty wonky to begin with. With the parametrization I've split out the tests and given preference to `xlsxwriter` if both that and `openpyxl` are installed with `.xlsx` files, which is what the original function was doing in a roundabout way